### PR TITLE
Adds option to use ICGS as orthogonalization method in BKS and LOBPCG

### PR DESCRIPTION
**Pull request by hkthorn**
@trilinos/anasazi 

## Motivation

The ICGS orthogonalization method has been an option in Anasazi, but has
not been available for some more commonly used eigensolvers, like BKS and
LOBPCG.  Similar to the study performed for Belos, it is of interest to
use this orthogonalization method to see if it improves performance on
more advanced architectures.

## Testing

Tests were modified to include ICGS orthogonalization.

Tests pass on:  OSX, serial, GCC 9.x

### CODE REVIEW

File: packages/anasazi/epetra/test/LOBPCG/cxx_main.cpp [hunk 1]
```diff
@@ -190,11 +196,13 @@ int main(int argc, char *argv[])
...
   MyPL.set( "Locking Tolerance", tol/10 );
   MyPL.set( "Full Ortho", fullOrtho );
+  MyPL.set( "Orthogonalization", ortho );
+
   if (testRecovery) {
     // initialize with bad P
     RCP<Anasazi::LOBPCGState<ScalarType,MV> > badstate = rcp( new Anasazi::LOBPCGState<ScalarType,MV>() );
-    Anasazi::SVQBOrthoManager<ScalarType,MV,OP> ortho(M);
-    ortho.normalize(*ivec,null);
+    Anasazi::SVQBOrthoManager<ScalarType,MV,OP> orthoM(M);
```
Comment (mhoemmen) [inline packages/anasazi/epetra/test/LOBPCG/cxx_main.cpp]: If users can specify the orthogonalization method, then why is SVQBOrthoManager hard-coded here?  I'm guessing that this a special case and not the main orthogonalization loop in LOBPCG.

File: packages/anasazi/src/AnasaziBlockKrylovSchurSolMgr.hpp [hunk 2]
```diff
@@ -388,7 +389,7 @@ BlockKrylovSchurSolMgr<ScalarType,MV,OP>::BlockKrylovSchurSolMgr(
 
   // which orthogonalization to use
   ortho_ = pl.get("Orthogonalization",ortho_);
-  if (ortho_ != "DGKS" && ortho_ != "SVQB") {
+  if (ortho_ != "DGKS" && ortho_ != "SVQB" && ortho_ != "ICGS") {
```
Comment (mhoemmen) [inline packages/anasazi/src/AnasaziBlockKrylovSchurSolMgr.hpp]: Anasazi doesn't have "IMGS" (while Belos does), right?

Comment (hkthorn) [reply packages/anasazi/src/AnasaziBlockKrylovSchurSolMgr.hpp]: Correct.  There isn't IMGS in Anasazi, only those three methods.  An orthogonalization factory would be preferable, but was never implemented in Anasazi.

File: packages/anasazi/epetra/test/LOBPCG/cxx_main.cpp (hunk 1 above)
Comment (hkthorn) [reply packages/anasazi/epetra/test/LOBPCG/cxx_main.cpp]: This is a special case where we are trying to create a bad subspace to make sure the recovery mechanism in the solver works.  The orthogonalization method is not critical to this part of the test.

File: packages/anasazi/src/AnasaziBlockKrylovSchurSolMgr.hpp (hunk 2 above)
Comment (mhoemmen) [reply packages/anasazi/src/AnasaziBlockKrylovSchurSolMgr.hpp]: Cool; thanks for explaining!

File: packages/anasazi/epetra/test/LOBPCG/cxx_main.cpp (hunk 1 above)
Comment (mhoemmen) [reply packages/anasazi/epetra/test/LOBPCG/cxx_main.cpp]: Cool; got it!

**Pull request merged**